mixer/src/Module.C

```diff
@@ -749,6 +753,41 @@ Module::Port::osc_control_change_cv ( float v, void *user_data )
 }
 
 
+int
+Module::Port::osc_control_update_signals ( void *user_data )
+{
+    Module::Port *p = (Module::Port*)user_data;
+
+    float f = p->control_value();
+
+    if ( p->unscaled_signal() )
+    {
+
+        p->unscaled_signal()->value_no_callback(f);
+    }
+
+    if ( p->scaled_signal() && p->hints.ranged)
+    {
+        // scale value to range.
+
+        float scale = p->hints.maximum - p->hints.minimum;
+        float offset = p->hints.minimum;
+        f =  ( f - offset ) / scale;
+
+        if ( f > 1.0f )
+            f = 1.0f;
+        else if ( f < 0.0f )
+            f = 0.0f;
+
+        p->scaled_signal()->value_no_callback(f);
+    }
+
+    Fl::unlock();
+
+    return 0;
+}
+
+
```

Comment:
Updates the signal's value according to the port's value. Called before sending reply to a query (value-less message)